src/components/TradeWidget/TokenRow.tsx

```diff
@@ -190,6 +202,17 @@ const TokenRow: React.FC<Props> = ({
     setValue(inputId, formatAmountFull(balance.exchangeBalance, balance.decimals, false))
   }
 
+  const enforcePrecision = useCallback(() => {
```

Comment:
No need to create this outside of `useEffect`

Reply:
I need it because this function is also used for `onChange` on the input.

Reply:
Missed that, my bad :sweat:

Reply:
Unforgivable!